### PR TITLE
pullTemplate: Update branch button

### PR DESCRIPTION
**Pull request by bharg**
pullTemplate: Update branch button

PULL_REQUEST_TEMPLATE.md now contains a warning about
the 'Update Branch' button

Fixes https://github.com/coala/coala/issues/3406

### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by jayvdb** [comment]
Could you run `git commit --amend` and change

1. the first line to "Pull template: Add branch button warning"
2. the last line to "Closes https://github.com/coala/coala/issues/3406"

And save that commit message.

You can check the commit message was changed using `git show`.

Then send the revision to this pull request using `git push -f myfork`

Grab us on gitter if you need help.

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Reviewing pull requests takes a lot of time and we're all volunteers. Please mak
 - [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
 - [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!
 
+> **After you submit your pull request, DO NOT click the 'Update Branch' button**
```
Comment (RohanVB) [inline PULL_REQUEST_TEMPLATE.md]: Do we want this as a quote? I think it'd be better in simple bold text

**Comment by RohanVB** [comment]
Please rebase, squash the extra commit and then push :)

**Comment by Mixih** [comment]
Good! there is also an issue about adding it to the docs if you're interested :)

**Comment by bharg** [comment]
I'm still not very clear with rebasing and squashing of commits. Will create a new request all together, if necessary.

**Comment by Shade5** [comment]
Are you working on a branch? If so, its easier to squash and rebase.
Pull latest commits on `master`.
Switch to your branch, reset to some old commit (make sure you save your change).
Then use `git rebase master`
Use apply your changes again, add and commit. You'll have one commit  that's rebased.

**Comment by adtac** [comment]
@bharg umm, why do you have 4 commits? After a review, *amend* your previous commit with `git commit --amend`, not create new commits on top.

Marking as WIP. If you have any questions, feel free to ask on the gitter channel.

**Comment by Mixih** [comment]
oh my, you seem to have included all of the other people's commits. Git reset to your first pull's sha, then reapply your changes, then try rebase again

**Pull request closed without merging**